contracts/settling_game/library/library_buildings.cairo

```diff
@@ -0,0 +1,224 @@
+        let (current_buildings_sqm) = get_current_built_buildings_sqm(current_buildings)
+
+        # Calculate requested building size
+        let (building_size) = get_building_size(building_id)
+
+        with_attr error_message("BUILDINGS: building size greater than buildable area"):
+            assert_le(building_size * quantity + current_buildings_sqm, buildable_units)
+        end
+
+        return ()
+    end
+
+    # gets building size for each
+    func get_building_size{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
+        building_id : felt
+    ) -> (buildable_area : felt):
+        alloc_locals
+        # TODO: Add other buildings sizes
+        let idx = building_id - 1
+
+        let (type_label) = get_label_location(building_area_sqm)
+
+        return ([type_label + idx])
+
+        building_area_sqm:
+        dw RealmBuildingsSize.House  # house
+        dw RealmBuildingsSize.StoreHouse  # storehouse
+        dw RealmBuildingsSize.Granary  # granary
+        dw RealmBuildingsSize.Farm  # farm
+        dw RealmBuildingsSize.FishingVillage  # fishing village
+        dw RealmBuildingsSize.Barracks  # barracks
+        dw RealmBuildingsSize.MageTower  # mage tower
+        dw RealmBuildingsSize.ArcherTower  # archer tower
+        dw RealmBuildingsSize.Castle  # castle
+    end
+
+    # gets buildable area for each
+    func get_realm_buildable_area{
+        syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr
+    }(cities : felt, regions : felt) -> (buildable_area : felt):
+        # Get buildable units
+        return (cities * regions + BASE_SQM)
+    end
+
+    # gets current built buildings
+    # pass building struct and return sqm
+    func get_current_built_buildings_sqm{
+        syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr
+    }(current_buildings : RealmBuildings) -> (size : felt):
+        # Get buildable units
+
+        let House = current_buildings.House * RealmBuildingsSize.House
+        let StoreHouse = current_buildings.StoreHouse * RealmBuildingsSize.StoreHouse
+        let Granary = current_buildings.Granary * RealmBuildingsSize.Granary
+        let Farm = current_buildings.Farm * RealmBuildingsSize.Farm
+        let FishingVillage = current_buildings.FishingVillage * RealmBuildingsSize.FishingVillage
+        let Barracks = current_buildings.Barracks * RealmBuildingsSize.Barracks
+        let MageTower = current_buildings.MageTower * RealmBuildingsSize.MageTower
+        let ArcherTower = current_buildings.ArcherTower * RealmBuildingsSize.ArcherTower
+        let Castle = current_buildings.Castle * RealmBuildingsSize.Castle
+
+        let size = House + StoreHouse + Granary + Farm + FishingVillage + Barracks + MageTower + ArcherTower + Castle
+
+        return (size)
+    end
+
+    # returns time to add
+    func get_final_time{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
```

Comment:
same comment about time, here it is a bit confusing. Maybe something like "integrity" or "life" like you used further on